R/dalys.R

```diff
@@ -0,0 +1,169 @@
+##' @param year_max maximum year
+##' @param vimc_dalys_only set to be TRUE if we are only interested in Ferrari, Li and LiST (201710gavi)
+##'
+##' @export
+calculate_dalys <- function(con, dalys_src, touchstone_name, year_min, year_max, vimc_dalys_only = TRUE){
+  ## [make temporary dalys parameter table]
+  dalys_parameters <- create_dalys_parameters(con, dalys_src=dalys_src, touchstone_name, vimc_dalys_only)
+  ## [make remaining life expectancy table] - this will take some time
+  life_table <- create_dalys_life_table(con, touchstone_name, year_min, year_max)
+  
+  sets <- unique(dalys_parameters$burden_estimate_set_id)
+  
+  dalys_out <- rep(list(NULL),length(sets))
+  for(i in seq_along(sets)){
+    burden_outcomes <- sql_in(unique(
+      dalys_parameters$burden_outcome_id[dalys_parameters$burden_estimate_set_id == sets[i]]
+    ),text_item = FALSE)
+    dalys_out[[i]] <- calculate_dalys1(con, life_table, sets[i], burden_outcomes, year_min, year_max)
+  }
+  # output
+  dat <- do.call(rbind, dalys_out)
+  dat$value <- dat$dalys
+  v <- DBI::dbGetQuery(con, "SELECT * FROM burden_outcome WHERE code = 'dalys'")
+  dat$burden_outcome <- v$id
+  cols <- c("burden_estimate_set", "country", "year", "burden_outcome", "value", "age")
+  dat <- dat[cols]
+}
+
+create_dalys_parameters <- function(con, touchstone_name = "201710gavi", dalys_src, vimc_dalys_only) {
+  ## subsetting those for which VIMC has to calculate DALYs - vimc_dalys = TRUE
+  if(vimc_dalys_only) {
+    dalys_src <- dalys_src[dalys_src$vimc_dalys, ]
+  }
+  ## transform average duration into years; N.B. duration will be compared with remaining life-expectancy
+  ## 1000 years of duration is funny, but helpful
+  dalys_src$duration <-  dalys_src$average_duration * duration_weighting(dalys_src$period)
+  
+  
+  i <- dalys_src$period == "year"
+  dalys_src$duration[i] <- dalys_src$average_duration[i]
+  i <- dalys_src$period == "month"
+  dalys_src$duration[i] <- dalys_src$average_duration[i] / 12
+  i <- dalys_src$period == "day"
+  dalys_src$duration[i] <- dalys_src$average_duration[i] / 365
+  
+  ## adjusted weight = proportion * diability_weight; this saves calculation later
+  dalys_src$adjusted_weight <- dalys_src$proportion * dalys_src$disability_weight
+  
+  ## find relevant responsibilities
+  sql <- read_sql(file_name = "dalys_calculation/responsibility.sql")
+  recipe <- DBI::dbGetQuery(con, sql, touchstone_name)
+  
+  dalys_src <- merge(dalys_src, recipe, by = c("disease", "modelling_group"), all.x = TRUE)
+  
+  dalys_src <- dalys_src[order(dalys_src$burden_estimate_set_id), ]
+  cols <- c("disease", "modelling_group", "scenario", "burden_estimate_set_id","burden_outcome_id", "code", "sub_condition", "duration", "adjusted_weight")
+  ## make a temporary table - dalys_parameters. This stores model-specific dalys parameters.
+  DBI::dbWriteTable(con, "dalys_parameters", dalys_src[cols], temporary = TRUE, append = TRUE, row.names = FALSE)
```

Comment:
remove `append = TRUE`

Reply:
When I remove `append = TRUE`, it returns error message requesting either `append = TRUE` or `overwrite = TRUE`. I now use `overwrite = TRUE`. 
In fact, `append = TRUE` did cause problem for the test data. `overwrite = TRUE` can be trusted. Therefore, I also updated the test data in `jenner-test-data` repo.